### PR TITLE
Litefs world db

### PR DESCRIPTION
**Pull request by jobez**
currently a draft because i had to hack some changes just to sanity check that litefs was correctly configured, will comment out what are temporary and will be lined up before taking out of draft

### CODE REVIEW

File: overlore/townhall/service.py [hunk 1]
```diff
@@ -5,12 +5,11 @@
 import signal
 from typing import Any
 
+import sqlean
```
Comment (jobez) [inline overlore/townhall/service.py:8]: temporarily connecting/querying world db in the entry point for debug convenience, can remove and add something simple that can be iterated on in a subsequent PR

File: overlore/townhall/service.py [hunk 2]
```diff
@@ -48,27 +47,38 @@ async def start() -> None:
     if TORII_WS is None:
         raise RuntimeError("Failure to provide WS url")
 
-    db = DatabaseHandler.instance().init()
+    DatabaseHandler.instance().init()
```
Comment (jobez) [inline overlore/townhall/service.py:50]: some changes were done by the linter in response to me commenting out code, in particular the removal of assignments

File: overlore/townhall/service.py [hunk 3]
```diff
@@ -48,27 +47,38 @@ async def start() -> None:
...
-        overlore_pulse,
-        torii_event_sub(TORII_WS, db.process_event, Subscriptions.COMBAT_OUTCOME_EVENT_EMITTED),
-        torii_event_sub(TORII_WS, db.process_event, Subscriptions.ORDER_ACCEPTED_EVENT_EMITTED),
-    )
+
+    conn = sqlean.connect(args.world_db)
+
+    print("sanity")
+
+    while True:
```
Comment (jobez) [inline overlore/townhall/service.py:65]: there was an issue with reliably establishing web socket connections, so I commented them out. then with those commented out, the overlore_pulse service wouldn't block on the gather call, so the program would just terminate on start.

that's why, temporarily, i commented it out and just put a while true loop and queried the world.db that the litefs service is meant to sync

File: overlore/townhall/service.py [hunk 4]
```diff
@@ -48,27 +47,38 @@ async def start() -> None:
...
+    # overlore_pulse,
+    # torii_event_sub(TORII_WS, db.process_event, Subscriptions.COMBAT_OUTCOME_EVENT_EMITTED),
+    # torii_event_sub(TORII_WS, db.process_event, Subscriptions.ORDER_ACCEPTED_EVENT_EMITTED),
+    # )
 
 
 # hardcode for now, when more mature we need some plumbing to read this off a config
 def main() -> None:
     asyncio.run(start())
+    print("wtf")
```
Comment (jobez) [inline overlore/townhall/service.py:81]: added this to sanity check that the program was just exiting early when the service stuff was started from docker. its possible whats happening is that an error is being thrown, but it isn't being logged correctly.

**Review by jobez — COMMENTED** [review]
initial explanatory review

**Comment by edisontim** [comment]
The way it connects looks really clean

**Review by edisontim — CHANGES_REQUESTED** [review]
Could you add something in the readme.md about the world_db pls?

**Comment by jobez** [comment]
> Could you add something in the readme.md about the world_db pls?

Would be easier when we merge the docker compose changes, so I can link out to it in the docker system files in eternums. Will just link out to it assuming those changes will get merged.

**Review by edisontim — APPROVED** [review]

**Pull request merged**